npm_and_yarn/Dockerfile

```diff
@@ -35,9 +37,6 @@ ENV NODE_EXTRA_CA_CERTS=/etc/ssl/certs/ca-certificates.crt
 
 # END: HACKY WORKAROUND FOR NPM GIT INSTALLS SPAWNING CHILD PROCESS
 
-# Our native helpers pull in yarn 1, so we need to reset the version globally
```

Comment:
Is this statement no longer true? Because the new ordering in this PR never resets the version after the native helpers run...

Reply:
Yeah, I think this statement was never true.

Only thing that should override a `corepack prepare <package_manager>@<version> --activate` should be a different `corepack prepare <package_manager>@<version> --activate` command, which I don't think installing native helpers does.

I think the original problem here was that the first `corepack prepare <package_manager>@<version> --activate` was being run as root, so the corepack binstubs were being installed in a location that the dependabot user won't use. That's why we needed to rerun the command again as the dependabot user.

I think I can extract this change to a separate PR to make sure it's good.

Reply:
This landed in https://github.com/dependabot/dependabot-core/pull/7160